tagid.go

```diff
@@ -4,105 +4,125 @@
 package swid
 
 import (
-	"encoding/hex"
 	"encoding/json"
 	"encoding/xml"
 	"errors"
 	"fmt"
 
 	"github.com/fxamacker/cbor/v2"
+	"github.com/google/uuid"
 )
 
-// TagID is the type of a tag identifier. Allowed formats (enforced via
-// checkTagID) are string or [16]byte
+// TagID is the type of a tag identifier. Allowed formats are string or
+// a valid universally unique identifier (UUID) as defined by RFC4122.
 type TagID struct {
 	val interface{}
 }
 
-// NewTagID returns a TagID initialized with the supplied value v
-// v is either a string or a [16]byte
 func NewTagID(v interface{}) *TagID {
-	if checkTagID(v) != nil {
+	switch t := v.(type) {
+	case string:
+		tagID, _ := string2TagID(t)
+		return tagID
+	case []byte:
+		tagID, _ := NewTagIDFromUUIDBytes(t)
+		return tagID
+	default:
 		return nil
 	}
-	return &TagID{v}
 }
 
-// String returns the value of the TagID as string. If the TagID has type
-// [16]byte the Base 16 encoding is returned
-func (t TagID) String() string {
-	switch v := t.val.(type) {
-	case string:
-		return v
-	case []byte:
-		return hex.EncodeToString(v)
-	default:
-		return "unknown type for tag-id"
+func string2TagID(s string) (*TagID, error) {
+	if tagID, err := NewTagIDFromUUIDString(s); err == nil {
```

Comment:
I am not clear as to why Line 36 is needed ? A received string should always be treated as "String" TAG ID???

Reply:
the reason for L36 is first we try to match the string against a UUID, otherwise we would not able to map SWID to CoSWID.

Reply:
Discussed this aspect and I am happy with the explaination!